numpy/core/fromnumeric.py

```diff
@@ -63,10 +82,46 @@ def _wrapfunc(obj, method, *args, **kwds):
     # in its class, but its signature is not identical to that
     # of NumPy's. This situation has occurred in the case of
     # a downstream library like 'pandas'.
-    except (AttributeError, TypeError):
+    except (AttributeError, TypeError) as e:
+        if not _is_compat_error(e, method):
+            raise
+
         return _wrapit(obj, method, *args, **kwds)
 
 
+def _wrapfunc2(obj, func, method, *args, **kwds):
+    # We perform this check because the ndarray implementation
+    # points to the implementation in this module. Thus, we would
+    # be going in a circle if we then called the ndarray method.
+    if type(obj) is not mu.ndarray:
+        try:
+            return getattr(obj, method)(*args, **kwds)
+
+        # An AttributeError occurs if the object does not have
+        # such a method in its class.
+
+        # A TypeError occurs if the object does have such a method
+        # in its class, but its signature is not identical to that
+        # of NumPy's. This situation has occurred in the case of
+        # a downstream library like 'pandas'.
+        except (AttributeError, TypeError) as e:
+            # In accordance with the API of these functions, if 'obj'
+            # is a sub-class of np.ndarray, and a non-default value for
+            # 'keepdims' is passed in, any exceptions thrown during
+            # the call will be raised.
+            if isinstance(obj, mu.ndarray) and 'keepdims' in kwds:
```

Comment:
Good question.  _Probably_ not, but it is a precaution in case there is someone out there who for some reason or another decides to invalidate the `ndarray` implementation via `AttributeError`.